src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -938,29 +960,34 @@ void useTerminalHooks(ConsoleProcessPtr cp)
 
    if (isZshShell())
    {
-      FilePath zDotDir =
-            session::options().rResourcesPath().completeChildPath("terminal/zsh");
+      core::FilePath zshDotDir = session::options().rResourcesPath().completeChildPath("terminal/zsh");
+      core::FilePath zshProfile = zshDotDir.completeChildPath(".zprofile");
 
-      cp->setenv("ZDOTDIR", zDotDir.getAbsolutePath());
+      // set ENV so that our terminal hooks are run
+      const char* env = ::getenv("ENV");
+      cp->setenv("_REALENV", env ? env : "<unset>");
+      cp->setenv("ENV", zshProfile.getAbsolutePath());
    }
+   
+   // make sure terminals see R temporary directory
+   // (older versions of R didn't set R_SESSION_TMPDIR; newer ones do)
+   cp->setenv("R_SESSION_TMPDIR", module_context::tempDir().getAbsolutePath());
 
    // set RSTUDIO_TERMINAL_HOOKS (to be sourced on startup by supported shells)
-   FilePath hooksPath =
-         session::options().rResourcesPath().completeChildPath("terminal/hooks");
-
+   FilePath hooksPath = session::options().rResourcesPath().completeChildPath("terminal/hooks");
    cp->setenv("RSTUDIO_TERMINAL_HOOKS", hooksPath.getAbsolutePath());
-   
 }
    
 void augmentTerminalProcess(ConsoleProcessPtr cp)
 {
-   // check whether terminal hooks are required
-   // (currently, only done if we want to place python on PATH)
-   bool needsTerminalHooks = augmentTerminalProcessPython(cp);
-   if (needsTerminalHooks)
+   if (prefs::userPrefs().terminalHooks())
+   {
+      if (!prefs::userPrefs().terminalPythonIntegration())
```

Comment:
Does this need to run `augmentTerminalProcessPython` if Python integration is enabled? It seems that `augmentTerminalProcessPython` used to set some env vars if Python integration was enabled.